main.md

```diff
@@ -981,6 +981,33 @@ MUST only be used with that server and no other.
 Developers should also take care to not confuse this nonce with the
 OpenID Connect [@OpenID.Core] ID Token nonce, should one also be present.
 
+# Authorization Code Binding to DPoP Key {#dpop_jkt}
+
+Binding the authorization code issued to the client's proof-of-possession key
+can enable end-to-end binding of the entire authorization flow.
+This specification defines the `dpop_jkt` authorization request parameter for this purpose.
+The value of the `dpop_jkt` authorization request parameter is the
+JSON Web Key (JWK) Thumbprint [@!RFC7638] of the proof-of-possession public key.
+
+When a token request is received, the authorization server computes the
+JWK thumbprint of the proof-of-possession public key in the DPoP proof
+and verifies that it matches the code challenge.
+If they do not match, it MUST reject the request.
+
+An example authorization request using the `dpop_jkt` authorization request parameter is:
+!---
+```
+ GET /authorize?response_type=code&client_id=s6BhdRkqt3&state=xyz
+     &redirect_uri=https%3A%2F%2Fclient%2Eexample%2Ecom%2Fcb
+     &dpop_jkt=NzbLsXh8uDCcd-6MNwXF4W_7noWXFZAfHkxZsRGC9Xs HTTP/1.1
+ Host: server.example.com
+```
+!---
+Figure: Authorization Request using the `dpop_jkt` Parameter
+
+Note that the `dpop_jkt` authorization request parameter MAY also be used
+in combination with PKCE [@RFC7636].
```

Comment:
Given that [best practices advise to always use PKCE](https://datatracker.ietf.org/doc/html/draft-ietf-oauth-security-topics-13#section-3.1.1), why not have the example include PKCE parameters, as this would constitute *recommended* usage (alternatively, shouldn't the "MAY" be a "SHOULD")?

If it is asserted that the security value of PKCE is made redundant by the use of dpop_jkt then this can this be explicitly stated for easier understanding, in a similar way that [BCP currently states that PKCE makes state redundant](https://datatracker.ietf.org/doc/html/draft-ietf-oauth-security-topics-13#section-3.1)?

Reply:
Good point.  I can add PKCE parameters to the example in the next draft.

Reply:
> If it is asserted that the security value of PKCE is made redundant by the use of dpop_jkt then can this please be explicitly stated for easier understanding, in a similar way that BCP currently states that PKCE makes state redundant?

What is your thinking on this - is there still security value from additional use of PKCE when dpop_jkt is used in this manner to bind the authorization code to the client?

Reply:
dpop_jkt provides a similar protection as client authentication when the client re-uses the dpop key, and provides a similar level of protection as PKCE if (and only if) the client choses a new dpop key for every request.

Reply:
To @danielfett 's point - dpop_jkt makes PKCE redundant only in the specific condition that a new dpop key is used for every authorization request.

Reply:
PKCE added to the example